src/runtime/uv/signal.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Sofia Rodrigues
+*/
+
+#include "runtime/uv/udp.h"
+
+namespace lean {
+#ifndef LEAN_EMSCRIPTEN
+
+using namespace std;
+
+typedef struct {
+    uv_signal_t signal;
+    lean_object* promise;
+} signal_req_t;
+
+/* Std.Internal.UV.Signal.wait (signum : Int) : IO (IO.Promise (Except IO.Error Unit)) */
+extern "C" LEAN_EXPORT lean_obj_res lean_uv_signal_wait(lean_obj_arg signum_obj, obj_arg /* w */) {
+    int signum = lean_scalar_to_int(signum_obj);
```

Comment:
I think you should assert here that `signum_obj` is actually a scalar or have `Std.Internal.UV.Signal.wait` take a `UIntX` instad of an `Int`.